Embedding two files with the same target directory

### Issue description
I am trying to embed two different files into the enclave. If their target path within the enclave share the same directory it throws an error when trying to run the executable.


### To reproduce
Steps to reproduce the behavior:

1. Configure enclave.json:
```
{
 "exe": "tags",
 "key": "private.pem",
 "debug": true,
 "heapSize": 512,
 "executableHeap": false,
 "productID": 1,
 "securityVersion": 1,
 "mounts": null,
 "env": null,
 "files": [
         {
            "source": "secret.txt",
            "target": "testdir/secret.txt"
        },
        {
            "source": "secret2.txt",
            "target": "testdir/secret2.txt"
        }
 ]
}
```
2. Build, sign and run the program.
3. Error:
```
EGo v1.0.1 (e1e48c9dbfdfd3cb2f2fda7602729162c9f762cc)
[erthost] loading enclave ...
[erthost] entering enclave ...
panic: mkdir testdir: file exists

goroutine 17 [running, locked to thread]:
main.ert_ego_premain(0x7fbbc14c6448, 0x7fbbc14c6440, 0xd408c000?, 0x0?, 0x1?)
        ego/premain/main.go:31 +0x1e5
````

### Additional info
From the error it seems like it is not checking whether the path already exists within the enclave before using mkdir.

--- a/ego/cmd/integration-test/enclave.json
+++ b/ego/cmd/integration-test/enclave.json
@@ -31,6 +31,10 @@
         {
             "source": "/tmp/ego-integration-test/file-host.txt",
             "target": "/path/to/file_enclave.txt"
+        },
+        {
+            "source": "/tmp/ego-integration-test/file-host.txt",
+            "target": "/path/to/even/deeper/nested/file_enclave.txt"
         }
     ]
 }

--- a/ego/premain/core/core.go
+++ b/ego/premain/core/core.go
@@ -76,6 +76,7 @@ func PreMain(payload string, mounter Mounter, fs afero.Fs, hostEnviron []string)
 			return err
 		}
 
+		// Write embedded files to file system
 		if err := writeFiles(config.Files, fs); err != nil {
 			return err
 		}
@@ -192,7 +193,7 @@ func writeFiles(files []config.File, fs afero.Fs) error {
 		if err != nil {
 			return err
 		}
-		if err := afs.MkdirAll(filepath.Dir(file.Target), 0); err != nil {
+		if err := afs.MkdirAll(filepath.Dir(file.Target), 0o700); err != nil {
 			return err
 		}
 		if err := afs.WriteFile(file.Target, buf, 0o600); err != nil {

